

rule all:
   input:
      "output/01vcf2fasta_gene",
      "output/02pi/loci_pi.txt"



##index_fasta: index genome
rule index_fasta:
   conda: "env.yml"
   input:
      "data/R0.fasta"
   output:
      "data/R0.fasta.fai"
   shell: """
      samtools faidx {input}
   """



##index_vcf: index vcf SNPs file
rule index_vcf:
   conda: "env.yml"
   input:
      snps="data/variants_merged_filt_SNPs.recode.vcf"
   output:
      snps="data/variants_merged_filt_SNPs.recode.vcf.gz.tbi"
   shell: """
      bgzip {input.snps}
      tabix {input.snps}.gz
   """



##vcf2fasta: obtain complete gene alleles in fasta for all samples
rule vcf2fasta:
   conda: "env.yml"
   input:
      fasta="data/R0.fasta",
      fai="data/R0.fasta.fai",
      gff="data/R0.gff",
      snps="data/variants_merged_filt_SNPs.recode.vcf.gz",
      vcfindex="data/variants_merged_filt_SNPs.recode.vcf.gz.tbi"
   output:
      directory("output/01vcf2fasta_gene")
   shell: """
      # https://github.com/santiagosnchez/vcf2fasta.git
      # see this thread if using GFF! https://github.com/santiagosnchez/vcf2fasta/issues/6 
      
      export PATH=$PATH:~/programs/vcf2fasta/

      vcf2fasta.py \
         --fasta {input.fasta}  \
         --vcf {input.snps}  \
         --gff {input.gff} \
         --feat gene \
         --addref  \
         --out output/01vcf2fasta
   """



##calculape_pi: calculate pi for each gene
rule calculape_pi:
   input:
      "output/01vcf2fasta_gene/"
   output:
      "output/02pi/loci_pi.txt"
   log:
      "logs/get_pi_per_file.log"
   shell: """
      Rscript scripts/get_pi_per_file.R {input} {output} > {log} 2>&1
   """



